wp/lib/bap_wp/src/precondition.ml

```diff
@@ -916,21 +886,23 @@ let visit_block (env : Env.t) (post : Constr.t) (blk : Blk.t) : Constr.t * Env.t
   let pre, env = blk |> Blk.elts ~rev:true |> compute_pre in
   (pre, Env.add_precond env (Term.tid blk) pre)
 
+(* Returns [true] if the node is not reachable from the start of the graph.
+   This is used to prune non-reachable subgraphs from the DFS. *)
+let unreachable_from_start (graph : Graphs.Ir.t) (start : Graphs.Ir.Node.t)
+    (node : Graphs.Ir.Node.t) : bool =
+  not (Graphlib.is_reachable (module Graphs.Ir) graph start node)
+
 let visit_graph (env : Env.t) (post : Constr.t)
```

Comment:
Do we need comments here? It's worth noting that if you skip nodes for which another node needs the precondition, this function may fail.

Reply:
Can do.